pkg/status/toolchaincluster.go

```diff
@@ -47,9 +47,15 @@ func GetToolchainClusterConditions(logger logr.Logger, attrs ToolchainClusterAtt
 	foundLastProbeTime := false
 	for _, condition := range toolchainCluster.ClusterStatus.Conditions {
 		if condition.Type == toolchainv1alpha1.ToolchainClusterReady {
-			lastProbeTime = condition.LastProbeTime
+			lastProbeTime = func() metav1.Time {
+				if condition.LastUpdatedTime != nil {
+					return *condition.LastUpdatedTime
+				}
+				return condition.LastProbeTime
```

Comment:
Could you please add/update the unit tests so it verifies that it give the LastUpdatedTime preference and if not set then it takes the LastProbeTime?
I mean, let's verify that it works when either from these two fields are not set. If I'm not mistaken, the current unit tests sets both fields

Reply:
added it